### ISSUE TITLE
Default revset for `jj git push` is imperfect

### ISSUE THREAD

**Issue by ilyagr** [martinvonz/jj#2434]
## Steps to Reproduce the Problem

1. jj branch c a
1. jj new
1. jj branch c b
2. jj git push --all
3. jj branch d b
4. jj branch s a

At this point, the repo should look like:

```
@ a@origin b
|
o b@origin
```

Now, do `jj git push`.

## Expected Behavior

The branch `a` gets deleted on the remote, `b` gets moved.

## Actual Behavior

Nothing happened.

This works as documented, as it pushes `remote_branches(remote="origin")..@`. However, I find this behavior against the spirit of what `jj git push` is supposed to do.

I'm not sure if there's a straightforward solution, though.

## Specifications

- Platform: Debian linux
- Version: recent nightly

**Comment by martinvonz** [comment]
Good point. I wonder why I didn't make it push all branches pointing into `::@` instead. It's not clear from https://github.com/martinvonz/jj/pull/2135/commits/2b24731e7821658abd219a22da4d09cba88e1688 if I considered that (it sounds like I didn't). Do you see a reason we shouldn't make that the default instead?

**Comment by arxanas** [comment]
@martinvonz doesn't that risk force-pushing the main branch or other unintentional branches?

**Comment by martinvonz** [comment]
> @martinvonz doesn't that risk force-pushing the main branch or other unintentional branches?

If you've moved it to be ancestor of `@`, yes. Do you think that's likely to happen? It pretty much requires the user to run `jj branch set main [-r <rev>]`, so the user probably knows if they've done it?

**Comment by arxanas** [comment]
Well, I would imagine that, by default, many people create commits/branches which are descendants of the main branch? That wouldn't require setting the main branch to anything in particular to have it as an ancestor.
